include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -486,6 +486,61 @@ class TensorProductMatrixSymmetricSumCollection
 
 
 
+/**
+ * A namespace with functions that create input for certain standard matrices
+ * for the classes TensorProductMatrixSymmetricSum and
+ * TensorProductMatrixSymmetricSumCache.
+ */
+namespace TensorProductMatrixCreator
+{
+  /**
+   * Boundary type that can be used in create_laplace_tensor_product_matrix();
+   */
+  enum LaplaceBoundaryType
+  {
+    dirichlet,
+    neumann,
+    internal_boundary,
+  };
+
+  /**
+   * Create 1D mass matrix and 1D derivative matrix for a scalar
+   * consant-coefficient
+   * Laplacian cell for a @p dim dimensional Cartesian cell. Its boundary types
+   * can be specified with @p bids. The cell extend (including the cell extend
+   * of each neighbor) can be specified via @p cell_extend. With @p n_overlap, an
+   * overlap with neighboring cells can be specified. The default value is one,
+   * which correspond to all matrix enties restricted to the cell-local DoFs.
+   */
+  template <int dim, typename Number>
+  std::pair<std::array<FullMatrix<Number>, dim>,
+            std::array<FullMatrix<Number>, dim>>
+  create_laplace_tensor_product_matrix(
+    const FiniteElement<1> &                            fe,
+    const Quadrature<1> &                               quadrature,
+    const dealii::ndarray<LaplaceBoundaryType, dim, 2> &bids,
+    const dealii::ndarray<double, dim, 3> &             cell_extend,
+    const unsigned int                                  n_overlap = 1);
+
+  /**
+   * Same as above but the boundary IDs are extracted from the given @p cell
+   * and are mapped to the boundary type via the sets @p dbcs and @p nbcs.
+   */
+  template <int dim, typename Number>
+  std::pair<std::array<FullMatrix<Number>, dim>,
+            std::array<FullMatrix<Number>, dim>>
+  create_laplace_tensor_product_matrix(
+    const typename Triangulation<dim>::cell_iterator &cell,
+    const std::set<types::boundary_id> &              dbcs,
```

Comment:
```suggestion
    const std::set<types::boundary_id> &              dirichlet_boundaries,
```
and similar for the Neumann case; what would you do in the Robin boundary case?

Reply:
I'll need to think about this.